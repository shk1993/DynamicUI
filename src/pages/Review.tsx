import * as React from 'react';
import ICurrentQuestion from '../interfaces/currentQuestion';

/**
 * Displays the current values in history as a preformatted view
 */
export default (props: ICurrentQuestion) => (
  <pre>{JSON.stringify(props.values, undefined, 2)}</pre>
);
