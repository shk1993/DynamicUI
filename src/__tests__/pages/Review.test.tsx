import * as React from 'react';
import { render } from 'react-testing-library';
import Review from '../../pages/Review';

describe('Review Page', () => {
  it('should render the review page and match the snapshot', () => {
    const { container } = render(
      <Review
        currentQuestion="review"
        values={{
          DE30: '06/05/2018',
          DE3: 'A9',
          DE15: 'awfeasdf',
          DE78: 'A252',
          DE81: 'A297',
          numOfLE: 'NO',
          DE21: ['A41'],
          DE2006: ['A1972', 'A1965'],
          DE522: 'A2248',
          DE2000: ['A2088', 'A2095'],
          DE2003: 'A1965',
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
