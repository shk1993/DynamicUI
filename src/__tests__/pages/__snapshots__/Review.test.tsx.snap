// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Page should render the review page and match the snapshot 1`] = `
<pre>
  {
  "DE30": "06/05/2018",
  "DE3": "A9",
  "DE15": "awfeasdf",
  "DE78": "A252",
  "DE81": "A297",
  "numOfLE": "NO",
  "DE21": [
    "A41"
  ],
  "DE2006": [
    "A1972",
    "A1965"
  ],
  "DE522": "A2248",
  "DE2000": [
    "A2088",
    "A2095"
  ],
  "DE2003": "A1965"
}
</pre>
`;
